gateway/cert.go

```diff
@@ -271,38 +272,40 @@ var tlsConfigCache = cache.New(60*time.Second, 60*time.Minute)
 var tlsConfigMu sync.Mutex
 
 func getTLSConfigForClient(baseConfig *tls.Config, listenPort int) func(hello *tls.ClientHelloInfo) (*tls.Config, error) {
-	// Supporting legacy certificate configuration
-	serverCerts := []tls.Certificate{}
-	certNameMap := map[string]*tls.Certificate{}
+	listenPortStr := strconv.Itoa(listenPort)
 
-	for _, certData := range config.Global().HttpServerOptions.Certificates {
-		cert, err := tls.LoadX509KeyPair(certData.CertFile, certData.KeyFile)
-		if err != nil {
-			log.Errorf("Server error: loadkeys: %s", err)
-			continue
+	return func(hello *tls.ClientHelloInfo) (*tls.Config, error) {
+		if config, found := tlsConfigCache.Get(hello.ServerName + listenPortStr); found {
+			return config.(*tls.Config).Clone(), nil
 		}
-		serverCerts = append(serverCerts, cert)
-		certNameMap[certData.Name] = &cert
-	}
 
-	for _, cert := range CertificateManager.List(config.Global().HttpServerOptions.SSLCertificates, certs.CertificatePrivate) {
-		if cert != nil {
-			serverCerts = append(serverCerts, *cert)
-		}
-	}
+		// if we have certificates ID's we don't cache the config if redis is not available yet.
+		noCaching := len(config.Global().HttpServerOptions.SSLCertificates) > 0 && !storage.Connected()
```

Comment:
Should not depend just on `SSLCertificates` field. If you use mTLS or similar you also should not cache if redis not enabled. So just change to `storage.Connected()`

Reply:
Also add a comment with explanation why it is needed